### ISSUE TITLE
CamelCase and PascalCase

### ISSUE THREAD

**Issue by dollar2048** [jsonmodel/jsonmodel#543]
Why CamelCase is deprecated now? 
I think it is a good idea to bring it back an add PascalCase support as well (or introduce ENUM with different cases).

So we can have something like this:
```
+ (JSONKeyMapper *)keyMapper
{
    return [JSONKeyMapper baseMapper:[JSONKeyMapper mapperForPascalCase]]
           withModelToJSONExceptions:@{
               @"estimatedCurrency": @"Estimate.Currency",
           }];
}
```
or 
```
+ (JSONKeyMapper *)keyMapper
{
    return [JSONKeyMapper mapperForPascalCase];
}
```
______________________________
because for now I am doing the following in the child class: 

```
+ (JSONKeyMapper *)keyMapper
{
    return [JSONKeyMapper baseMapper:[super keyMapper]
           withModelToJSONExceptions:@{
               @"estimatedCurrency": @"Estimate.Currency",
           }];
}

```
and in the super class:
```
+ (JSONKeyMapper *)keyMapper
{
    [JSONKeyMapper mapperForSnakeCase];
    JSONModelKeyMapBlock toJSON = ^NSString *(NSString *keyName)
    {
        NSString *capitalizedString = @"";

        if (keyName.length > 0)
            capitalizedString = [keyName stringByReplacingCharactersInRange:NSMakeRange(0, 1)
                                                                 withString:[[keyName substringToIndex:1] uppercaseString]];

        return capitalizedString;
    };

    return [[JSONKeyMapper alloc] initWithModelToJSONBlock:toJSON];
}
```

**Comment by billinghamj** [comment]
`camelCase` in the model to/from `snake_case` in the JSON is still supported :) I just renamed the method to make it more clear & follow the other naming convention. Just use it as follows:

```objc
+ (JSONKeyMapper *)keyMapper
{
  return [JSONKeyMapper mapperForSnakeCase];
}
```

**Comment by billinghamj** [comment]
The upper-case/lower-case option has been removed entirely as it didn't really work the best way. I'm not inclined to add this back, but we're open to suggestions as to how it could work better.

Are you looking for an option which would transform from `camelCase` in the model to/from `PascalCase` in the JSON? e.g.:

```json
{
  "SomeField": "blah"
}
```

```objc
@interface MyClass
@property (nonatomic) NSString *someField;
@end
```

**Comment by dollar2048** [comment]
@billinghamj Yes. correct. I am getting data in `PascalCase` format.
And would like to set it up in any superclass.

**Comment by billinghamj** [comment]
Great. I will add a `mapperForTitleCase` option.

**Comment by dollar2048** [comment]
I would be great.

**Comment by billinghamj** [comment]
`mapperForTitleCase` is now included in v1.6.0 :)
